File: Classe para lista/Java/Lista/lista.java
public class lista {
    private int nElementos, vetor[];

    public lista() {
        this.nElementos = 0;
        this.vetor = new int[20];
    }
    public void insere(int value){
        if(nElementos < 20){
            this.vetor[this.nElementos] = value;
            this.nElementos++;
        }
        else {
            System.out.println("Número máximo de elementos atingidos\n");
        }
    }  // Deve inserir um elemento na lista (na frente ou no final, tanto faz)
    public void retira(){
        if(nElementos > 0){
            this.vetor[this.nElementos] = -1;
            this.nElementos--;
        }
        else {
            System.out.println("Impossível retirar mais elementos\n");
        }
    } // Deve retirar um elementa da lista (do final ou da frente, tanto faz)
    public int first(){
        if(this.vetor[0] != -1){
            return vetor[0];
        }
        else {
            System.out.println("Lista vazia\n"); 
            return -1;
        }
    } // Retorna o valor armazenado no início da lista
    public int last(){
        if(this.vetor[0] != -1){
            return vetor[this.nElementos];
        }
        else {
            System.out.println("Lista vazia\n"); 
            return -1;
        }
    } // Retorna o valor armazenado no final da lista
    public int size(){
        return this.nElementos;
    } // Retorna o número de elementos da lista
}

File: Classe para lista/Python/Lista/lista.py
class Lista:
    def __init__(self):
        self.nElementos = 0
        self.vetor = []

    def insere(self, value):
        if self.nElementos < 20:
            self.vetor[self.nElementos] = value
            self.nElementos = self.nElementos + 1
        else:
            print("Número máximo de elementos atingidos\n")

    def retira(self):
        if self.nElementos > 0:
            self.vetor[self.nElementos] = -1
            self.nElementos = self.nElementos - 1
        else:
            print("Impossível retirar mais elementos\n")

    def first(self):
        if self.vetor[0] != -1:
            return self.vetor[0]
        else:
            print("Lista vazia\n")
            return -1
        
    def last(self):
        if self.vetor[0] != -1:
            return self.vetor[self.nElementos - 1]
        else:
            print("Lista vazia\n")
            return -1

    def size(self):
        return self.nElementos

        
